qiskit/aqua/operators/converters/circuit_sampler.py

```diff
@@ -0,0 +1,318 @@
+    """
+    def __init__(self,
+                 backend: Union[BaseBackend, QuantumInstance] = None,
+                 statevector: Optional[bool] = None,
+                 param_qobj: bool = False,
+                 attach_results: bool = False) -> None:
+        """
+        Args:
+            backend: The quantum backend or QuantumInstance to use to sample the circuits.
+            statevector: If backend is a statevector backend, whether to replace the
+                CircuitStateFns with DictStateFns (from the counts) or VectorStateFns (from the
+                statevector). ``None`` will set this argument automatically based on the backend.
+            param_qobj: (TODO, not yet available) Whether to use Aer's parameterized Qobj
+                capability to avoid re-assembling the circuits.
+            attach_results: Whether to attach the data from the backend ``Results`` object for
+                a given ``CircuitStateFn``` to an ``execution_results`` field added the converted
+                ``DictStateFn`` or ``VectorStateFn``.
+
+        Raises:
+            ValueError: Set statevector or param_qobj True when not supported by backend.
+        """
+        self._quantum_instance = backend if isinstance(backend, QuantumInstance) else\
+            QuantumInstance(backend=backend)
+        self._statevector = statevector if statevector is not None \
+            else self.quantum_instance.is_statevector
+        self._param_qobj = param_qobj
+        self._attach_results = attach_results
+
+        self._check_quantum_instance_and_modes_consistent()
+
+        # Object state variables
+        self._last_op = None
+        self._reduced_op_cache = None
+        self._circuit_ops_cache = {}
+        self._transpiled_circ_cache = None
+        self._transpile_before_bind = True
+        self._binding_mappings = None
+
+    def _check_quantum_instance_and_modes_consistent(self) -> None:
+        """ Checks whether the statevector and param_qobj settings are compatible with the
+        backend
+
+        Raises:
+            ValueError: statevector or param_qobj are True when not supported by backend.
+        """
+        if self._statevector and not is_statevector_backend(self.quantum_instance.backend):
+            raise ValueError('Statevector mode for circuit sampling requires statevector '
+                             'backend, not {}.'.format(self.quantum_instance.backend))
+
+        if self._param_qobj and not is_aer_provider(self.quantum_instance.backend):
+            raise ValueError('Parameterized Qobj mode requires Aer '
+                             'backend, not {}.'.format(self.quantum_instance.backend))
+
+    @property
+    def backend(self) -> BaseBackend:
+        """ returns backend
+
+        Returns:
+             The backend used by the CircuitSampler
+        """
+        return self.quantum_instance.backend
+
+    @backend.setter
+    def backend(self, backend: BaseBackend) -> None:
```

Comment:
I think if you pass a backend into the constructor, you expect to see it in the properties. I've changed this to mimic the QuantumAlgorithm structure but added a simple getter and setter for backend.

Reply:
Ok, of course this called be called quantum_instance as we have in the algos for consistency here too, allowing a backend or quantum instance

Reply:
I think it's nice to use backend because it's what users recognize, and if they know more about Aqua they know they can pass in the QuantumInstance. This isn't a QuantumAlgorithm, so I think it's fine to have it's own expectation

Reply:
Well this is part of Aqua! I would usually go for consistency but if feel strongly that its fine the way it is...